drivers/eeprom/eeprom_mchp_xec.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2022 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_eeprom
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/eeprom.h>
+#include <soc.h>
+
+#ifdef CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+#endif
+
+#define LOG_LEVEL CONFIG_EEPROM_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eeprom_xec);
+
+/* EEPROM Mode Register */
+#define XEC_EEPROM_MODE_ACTIVATE			BIT(0)
+
+/* EEPROM Status Register */
+#define XEC_EEPROM_STS_TRANSFER_COMPL			BIT(0)
+
+/* EEPROM Execute Register - Transfer size bit position */
+#define XEC_EEPROM_EXC_TRANSFER_SZ_BITPOS		(24)
+
+/* EEPROM Execute Register - Commands */
+#define XEC_EEPROM_EXC_CMD_READ			0x00000
+#define XEC_EEPROM_EXC_CMD_WRITE		0x10000
+#define XEC_EEPROM_EXC_CMD_READ_STS		0x20000
+#define XEC_EEPROM_EXC_CMD_WRITE_STS	0x30000
+
+/* EEPROM Execute Register - Address mask */
+#define XEC_EEPROM_EXC_ADDR_MASK		0x7FFU
+
+/* EEPROM Read/Write Transfer Size */
+#define XEC_EEPROM_TRANSFER_SIZE_READ	32
+#define XEC_EEPROM_TRANSFER_SIZE_WRITE	32
```

Comment:
If intention is to read / write full page, then these two macros seem optional, better just use `XEC_EEPROM_PAGE_SIZE` , or assign them with `XEC_EEPROM_PAGE_SIZE` and move macro definition above these 2.

Reply:
updated